stan/math/rev/core/var.hpp

```diff
@@ -295,5 +295,18 @@ class var_value {
 using var = var_value<double>;
 
 }  // namespace math
+
+/**
+ * Template specialization defining the scalar type of
+ * values stored in var_value.
+ *
+ * @tparam T type to check.
+ * @ingroup type_trait
+ */
+template <typename T>
+struct scalar_type<math::var_value<T>> {
+  using type = math::var_value<scalar_type_t<T>>;
+};
```

Comment:
So this is kind of confusing to me. What's the logic on returning back `var_value<double>` for a `var_value<MatrixXd>`?

Reply:
Well, `scalar_type` always was one element from the container. To capture all the data from present at particular index of `var_value<MatrixXd>` we need to have both value and adjoint. The scalar type containing that is `var_value<double>`. 

For example I would expect indexing of `var_value<MatrixXd>` to return `var_value<double>` once we implement that. Returning just a value or just adjoint would be the confusing thing.